routes/error_handler.go

```diff
@@ -0,0 +1,34 @@
+package routes
+
+import (
+	"net/http"
+
+	raven "github.com/getsentry/raven-go"
+	"github.com/gocardless/draupnir/routes/chain"
+	"github.com/prometheus/common/log"
+)
+
+func NewErrorHandler(logger log.Logger) chain.TerminatingMiddleware {
+	return func(next chain.Handler) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			err := next(w, r)
+			if err != nil {
+				logger.With("http_request", r).Error(err.Error())
+				RenderError(w, http.StatusInternalServerError, internalServerError)
+			}
+		}
+	}
+}
+
+func NewSentryErrorHandler(logger log.Logger, sentry *raven.Client) chain.TerminatingMiddleware {
+	return func(next chain.Handler) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			err := next(w, r)
+			if err != nil {
+				logger.With("http_request", r).Error(err.Error())
```

Comment:
Do we want to do this? I wasn't sure if we did, I was only doing it because the sentry logrus hook needed it.

It might be a good idea but want us to decide that ourselves instead of keeping it for status quo!

Reply:
I would like some way of connecting error logs to the request that caused them. Maybe we’re better off with a request id, but I thought this was a good enough first pass